### PR TITLE
Add OpRtnReg

### PR DESCRIPTION
**Pull request by tymm**
I couldn't find an expression that makes use of `opRtnReg`.
Therefore I just made up a test case:
```
  litvec:
    0:   100
  codevec:
    0:   liti 0,rslt
    1:   rtn/reg/nxt, trgt
```

`rtn/reg/nxt, trgt` corresponds to `OpRtnReg(next = true, reg = 1)` in Roscala.
`OpRtnReg(next = true, reg = 1)` copies the content of `state.ctxt.rslt` to `state.ctxt.trgt`.

### CODE REVIEW

File: roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala [hunk 1]
```diff
@@ -496,4 +496,27 @@ class TransitionSpec extends FlatSpec with Matchers {
...
   }
+
+  "OpRtnReg" should "copy ctxt.rslt to given register" in {
+
+    /**
+      * litvec:
+      *   0:   100
+      * codevec:
+      *   0:   liti 0,rslt
+      *   1:   rtn/reg/nxt
```
Comment (KentShikama) [inline roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Should be `* 1: rtn/reg/nxt 1`

Comment (KentShikama) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: Or I guess `* 1: rtn/reg/nxt trgt`

Comment (KentShikama) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: On further thought, it doesn't really make sense to specify the returning register of a codevec as that is the responsibility of the caller. Maybe

```
push/alloc 1
liti 0, rslt
rtn/reg/nxt, rslt
rtn/nxt
```
In other words the `rtn/reg/nxt, rslt` is effectively a no-op.

Comment (KentShikama) [reply roscala/src/test/scala/coop/rchain/rosette/TransitionSpec.scala]: As in the above is equivalent to if you did

```
liti 0, rslt
rtn/nxt
```

**Comment by tymm** [comment]
Added comment that opcode sequence is made up and changed PR summary.

**Pull request merged**